server/lib/src/idm/application.rs

```diff
@@ -0,0 +1,631 @@
+        self.inner.commit();
+    }
+}
+
+impl ApplicationResourceServers {
+    pub fn read(&self) -> ApplicationResourceServersReadTransaction {
+        ApplicationResourceServersReadTransaction {
+            inner: self.inner.read(),
+        }
+    }
+
+    pub fn write(&self) -> ApplicationResourceServersWriteTransaction {
+        ApplicationResourceServersWriteTransaction {
+            inner: self.inner.write(),
+        }
+    }
+}
+
+impl TryFrom<Vec<Arc<EntrySealedCommitted>>> for ApplicationResourceServers {
+    type Error = OperationError;
+
+    fn try_from(value: Vec<Arc<EntrySealedCommitted>>) -> Result<Self, Self::Error> {
+        let application_rs = ApplicationResourceServers {
+            inner: CowCell::new(ApplicationRSInner {
+                rs_set: HashMap::new(),
+            }),
+        };
+
+        let mut application_rs_wr = application_rs.write();
+        application_rs_wr.reload(value)?;
+        application_rs_wr.commit();
+        Ok(application_rs)
+    }
+}
+
+impl<'a> IdmServerAuthTransaction<'a> {
+    pub async fn application_auth_ldap(
+        &mut self,
+        lae: &LdapApplicationAuthEvent,
+        ct: Duration,
+    ) -> Result<Option<LdapBoundToken>, OperationError> {
+        let entry = self.qs_read.internal_search_uuid(lae.target)?;
+
+        let within_valid_window = Account::check_within_valid_time(
+            ct,
+            entry
+                .get_ava_single_datetime(Attribute::AccountValidFrom)
+                .as_ref(),
+            entry
+                .get_ava_single_datetime(Attribute::AccountExpire)
+                .as_ref(),
+        );
+
+        if !within_valid_window {
+            security_info!("Account has expired or is not yet valid, not allowing to proceed");
+            return Err(OperationError::SessionExpired);
+        }
+
+        let account = Account::try_from_entry_ro(&entry, &mut self.qs_read).map_err(|e| {
+            admin_error!("Failed to search account {:?}", e);
+            e
+        })?;
+
+        if account.is_anonymous() {
+            return Err(OperationError::InvalidUuid);
+        }
+
+        let application = self
+            .apprs
+            .inner
+            .rs_set
+            .get(&lae.application)
+            .ok_or_else(|| {
+                admin_info!("Application {:?} not found", lae.application);
+                OperationError::NoMatchingEntries
+            })?;
+
+        // Check linked group membership
+        let linked_group_uuid = match application.linked_group {
+            Some(u) => u,
+            None => {
+                admin_warn!(
+                    "Application {:?} does not have a linked group.",
+                    application.name
+                );
+                return Ok(None);
+            }
+        };
+
+        // TODO: Should we check if the group still exists or just trust in ref. integrity?
+
+        if !account.groups.iter().any(|e| e.uuid() == linked_group_uuid) {
```

Comment:
Probably easier to just check the entries memberOf directly? I think there is a better way than an iter() here.

Reply:
Ok, fixed.